src/wallet/api.rs

```diff
@@ -1237,6 +1264,212 @@ impl Wallet {
         Ok(descriptors_to_import)
     }
 
+    pub fn spend_coins(
+        &self,
+        coins: &Vec<&(ListUnspentResultEntry, UTXOSpendInfo)>,
+        destination: Destination,
+        feerate: Option<f64>,
```

Comment:
As the api.rs is already quite large, lets move this into `direct_spend`, that's pretty empty now. 

Can also move all other spend APIs into this module, contract spends, fidelity spend, swapcoin spend etc. 

And lets rename it to `spend.rs`. Theres no "indirect spend".